packages/flutter/lib/src/material/date_picker.dart

```diff
@@ -220,14 +222,21 @@ class DayPicker extends StatelessWidget {
     }).toList(growable: false);
   }
 
+  int _daysInMonth(int year, int month) {
+    if (month == DateTime.FEBRUARY) {
+      final bool isLeapYear = (year % 4 == 0) && (year % 100 != 0) || (year % 400 == 0);
+      if (isLeapYear)
+        return 29;
+    }
+    return _kDaysInMonth[month - 1];
+  }
```

Comment:
this is entirely up to you but i would be tempted to make _kDaysInMonth[1] have the value -1, and then make _daysInMonth return 29 if isLeapYear is true and 28 otherwise. That way anyone tempted to use _kDaysInMonth would immediately find that they'd forgotten about leap years. Also maybe add a comment next to _kDaysInMonth pointing to _daysInMonth.